// pages/auth-routes/Auth.js
import React,{useState, useEffect} from 'react'
import {Link} from 'react-router-dom'
import axios from 'axios'

const url =  'https://conduit.productionready.io/api/users/login'

export  function Auth() {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [isSubmitting, setIsSubmitting] = useState(false)

  const handleSubmit = e => {
    e.preventDefault();
    setIsSubmitting(true)
  }

  useEffect(() => {
    if(!isSubmitting) {
      console.log(isSubmitting)
     return
    }
    console.log(isSubmitting)

    axios(url, {
      method: 'post',
      data: {
        user: {
          email: 'emalkj.@..',
          password: '131'
        }
      }
    })
    .then(res=>{
      console.log(res)
    }
    )
    .catch(error=>{
      console.log(error)
      setIsSubmitting(false)
    })
  } )

  return (
    <div className='auth-page'>
     <div className='container'>
      <div className='row'>
        <div className='col-md-6 offset-md-3 col-xs-12'>
          <h1 className='login'>Login</h1>
          <p className='text-xs-center'>
            <Link to='register'>Need an account?</Link>
          </p>
          <form onSubmit={handleSubmit}>
            <fieldset>
              <fieldset className='form-group'>
              <input type='text'
                type='email'
                className='form-control form-control-lg'
                placeholder='Email'
                onChange={e=>setEmail(e.target.value)}
                value={email}
              />
              </fieldset>
              <fieldset className='form-group'>
              <input 
                type='password'
                type='password'
                className='form-control form-control-lg'
                placeholder='Password'
                value={password}
                onChange={e=> setPassword(e.target.value)}
              />
              </fieldset>
              <button 
                className='btn bnt-lg btn-primary pull-xs-right' 
                disabled={isSubmitting}
                type='submit'>
                Sign In
              </button>
            </fieldset>
          </form>
        </div>
      </div>
     </div>
    </div>
  )
}
